new_study_assistant.py

```diff
@@ -0,0 +1,624 @@
+    studies, study_attributes = get_attribute_data(all_cBioPortalStudies)
+else:
+    list_of_files = find_data_files(datahub_path)
+    study_attributes, attribute_value_data = load_data_from_files(list_of_files, datahub_path)
+study_names = get_study_names(study_attributes)
+study_data_combined = study_attributes_to_df(study_attributes)
+
+
+#choose a study to serve as a "new" study
+if random_study or specific_study is not None:
+    if random_study:
+        test_study = random.choice(study_names)
+    else:
+        test_study = specific_study
+    print "Study being analyzed: " + test_study
+    study_to_drop = test_study
+    study_data_combined = drop_study(test_study, study_data_combined)
+    test_study_data, test_study_clin_attributes = get_attribute_data([test_study])
+    test_study_attribute_data = test_study_data[0][1]
+    test_study_attribute_names = test_study_attribute_data['clinicalAttributeId'].values
+    test_study_patient_attribute_values = get_clinical_data_values([test_study], attribute_type="PATIENT")
+    test_study_sample_attribute_values = get_clinical_data_values([test_study], attribute_type="SAMPLE")
+    processed_test_study_patient_attribute_values = process_new_study_data(test_study_patient_attribute_values[0][1])
+    processed_test_study_sample_attribute_values = process_new_study_data(test_study_sample_attribute_values[0][1])
+    new_study_clinical_attribute_values = pd.concat([processed_test_study_patient_attribute_values,
+                                                     processed_test_study_sample_attribute_values],
+                                                     axis=0).fillna(value=0)
+
+#otherwise read in data from a new study
+else:
+    print "Study being analyzed: " + new_study_path
+    new_study_data = pd.read_table(new_study_path)
+    test_study_attribute_names = get_new_study_attributes(new_study_data)
+    if study_to_drop is not None:
+        study_data_combined = drop_study(study_to_drop, study_data_combined)
+    new_study_clinical_attribute_values = process_new_study_data(new_study_data)
+    
+#check matching attributes (via name only)
+exact_matching_attributes = np.intersect1d(test_study_attribute_names, list(study_data_combined))
+non_matching_attributes = np.setdiff1d(test_study_attribute_names, list(study_data_combined))
+possible_matches = find_attribute_name_matches(study_data_combined, non_matching_attributes)
+
+#check matching attributes based on data values
+if api_flag:
+    patient_data = get_clinical_data_values(all_cBioPortalStudies, attribute_type="PATIENT")
+    patient_data_processed = process_clinical_data(patient_data, study_to_drop)
+    sample_data = get_clinical_data_values(all_cBioPortalStudies, attribute_type="SAMPLE")
+    sample_data_processed = process_clinical_data(sample_data, study_to_drop)
+    combined_PS_attribute_data = pd.concat([patient_data_processed, sample_data_processed], axis=0).fillna(value=0)
```

Comment:
so patient data analysis is done separately from sample data analysis (when we know the category of attributes because we are using the api to get our own annotated studies). I think that makes sense.